include/seqan3/search/fm_index/fm_index.hpp

```diff
@@ -78,7 +79,19 @@ class bi_fm_index_iterator;
  * ### Running time / Space consumption
  *
  * \f$SAMPLING\_RATE = 16\f$
- * \f$\Sigma\f$: alphabet_size<char_type> where char_type is the seqan3 alphabet type (e.g. dna4 has an alphabet size of 4)
+ * \f$\Sigma\f$: alphabet_size<char_type> where char_type is the seqan3 alphabet type (e.g. dna4 has an alphabet size of 4).
+ *
+ * For an index over a text collection a delimiter is added inbetween the texts. This causes sigma to increase by 1.
+ * \attention For any alphabet, the symbol with rank 255 is not allowed to occur in the text. Addtionally,
+ *            rank 254 cannot occur when indexing text collections.
+ *
+ * \if DEV
+ * Rank 255 is generally not allowed. When constructing the index, we increase the rank of every letter by 1, since the
+ * SDSL uses 0 as a sentinel. Incrementing 255 by 1 will cause the rank (uint8_t) to overflow to 0.
+ * For text collections, 255 is reserved as delimiter between the individual texts. Therefore the letter with rank 254
+ * cannot occur in the text.
+ * \endif
```

Comment:
are there any sanity checks for this in the code? When you do the copying you could actually check the value and throw if it is out-of-range.